components/server-side-render/index.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	Component,
+	RawHTML,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+
+export class ServerSideRender extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			response: {},
+			attributes: props,
+		};
+	}
+
+	componentDidMount() {
+		this.getOutput();
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( JSON.stringify( nextProps ) !== JSON.stringify( this.props ) ) {
```

Comment:
What about using `isEqual` from lodash?

Reply:
1. I was going to comment that there was room to figure out how aggressive this component should be with refetches and ask if a [shallow equality](https://github.com/WordPress/gutenberg/pull/5616) check could make any sense. However, by design, block attributes should (until abused) retain a very simple shape. `isEqual` should do it.

2. More importantly, it's not clear to me why there is state synchronization at all. Perhaps this is a relic of a previous solution? `response` appears to be the only datum requiring local state. Component props should be fed directly into `getOutput`. If the goal is to trigger a refetch and a rerender, the following should be enough:

```js
componentWillReceiveProps( nextProps ) {
  if ( ! isEqual( nextProps, this.props ) ) {
    this.getOutput();
  }
}
```